### PR TITLE
Move the GCC and CLR Tools Unit Tests legs to use the global build template

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Currently the GCC job and the CoreCLR Tools Unit Tests job both use the CoreCLR build job template. These two jobs function quite differently than the rest of the jobs that go through this template. This PR moves these two jobs to use the global-build-job template and removes the various separate paths in the CoreCLR build job template that were only for these two jobs.

### CODE REVIEW

**Review by sbomer — APPROVED** [review]
Nice, thanks!

File: eng/pipelines/coreclr/templates/build-job.yml [hunk 1]
```diff
@@ -262,8 +232,8 @@ jobs:
           isOfficialBuild: ${{ parameters.signBinaries }}
           timeoutInMinutes: 30
 
-    # Builds using gcc are not tested, and clrTools unitests do not publish the build artifacts
-    - ${{ if and(ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, 'clrTools'), ne(parameters.disableClrTest, true)) }}:
+    # clrTools unitests do not publish the build artifacts
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply eng/pipelines/coreclr/templates/build-job.yml:235]:
```suggestion
```

File: eng/pipelines/runtime.yml [hunk 2]
```diff
@@ -331,13 +336,16 @@ extends:
...
-          jobTemplate: /eng/pipelines/coreclr/templates/build-job.yml
+          jobTemplate: /eng/pipelines/common/global-build-job.yml
           buildConfig: checked
           platforms:
           - linux_x64
           jobParameters:
-            testGroup: clrTools
             timeoutInMinutes: 120
+            nameSuffix: CLR_Tools_Tests
+            buildArgs: -s clr.aot+clr.iltools+libs+clr.toolstests -c $(_BuildConfig) -test
```
Comment (hoyosjs) [inline eng/pipelines/runtime.yml:346]: Is AOT for crossgen?

File: eng/pipelines/common/templates/runtimes/build-runtime-tests.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  liveLibrariesBuildConfig: ''
+  testBuildArgs: ''
+  #arcade-specific parameters
+  condition: always()
+  continueOnError: false
+  displayName: ''
+  timeoutInMinutes: ''
+  enableMicrobuild: ''
+  gatherAssetManifests: false
+  shouldContinueOnError: false
```
Comment (hoyosjs) [inline eng/pipelines/common/templates/runtimes/build-runtime-tests.yml:15]: If the template doesn't seem to use these, do we need to define them?

File: eng/pipelines/runtime.yml (hunk 2 above)
Comment (jkoritzinsky) [reply eng/pipelines/runtime.yml:346]: AOT is for the ILCompiler.Compiler.Tests suite for NativeAOT.

**Review by hoyosjs — APPROVED** [review]

File: eng/pipelines/common/templates/runtimes/build-runtime-tests.yml (hunk 3 above)
Comment (jkoritzinsky) [reply eng/pipelines/common/templates/runtimes/build-runtime-tests.yml:15]: We can remove them, but I was keeping consistency with our other templates. In another PR I’ll go through and clean up some of these across all of our extra-steps templates and make the parameter lists consistent.

**Comment by jkoritzinsky** [comment]
Failures are all known. Merging this in.

**Pull request merged**